src/librustc/mir/interpret/allocation.rs

```diff
@@ -198,7 +198,6 @@ impl<'tcx, Tag: Copy, Extra: AllocationExtra<Tag>> Allocation<Tag, Extra> {
         align: Align,
         check_defined_and_ptr: bool,
     ) -> EvalResult<'tcx, &[u8]> {
-        assert_ne!(size.bytes(), 0, "0-sized accesses should never even get a `Pointer`");
```

Comment:
I'm fine with keeping the assertions. I am also fine with removing them.

The important part is that we don't short-circuit for size zero.

Reply:
I can turn them into debug assertions. There's a CI runner with debug assertions on.

Reply:
But I checked, the code works fine even if given zero sized args

Reply:
I am fine either way.

Reply:
I'll leave them out. No need to check an invariant that doesn't actually exist